### DISCUSSION TITLE
Can't log in with GitHub on sites on Firefox mobile

### DISCUSSION THREAD

**Discussion by elibroftw** [community/community#70181] [Mobile]
When I go to log into a site with GitHub, for example foss.heptapod.net, I enter my username and password, click sign in, and then the GitHub app opens up and gives me 'something went wrong' instead of a more useful error message.

![Screenshot_20231015-193500.png](https://github.com/community/community/assets/21298211/8afd8d8b-21fc-4998-8aea-b38655b7a1a9)

When I go back to Firefox I just see this:

![Screenshot_20231015-194022.png](https://github.com/community/community/assets/21298211/d3199720-f26e-4687-831c-71e05c16fd4e)

This issue has been present for months, I just felt like reporting it today since I had the time.

Still applicable in June 2025!

**Comment by badergithub** [comment]
[]()

**Comment by elisealix22** [comment]
Hi @elibroftw thank you so much for the report and for including screenshots! One question for you: are you seeing the something went wrong error immediately after signing in? The reason I ask is that screen doesn't look to be on our home screen since there aren't any bottom tabs & there is a back button. Are you navigating to a screen after signing in and seeing the error? Or are you being pushed to that screen directly?

**Reply by elibroftw** [reply]
Pushed

**Reply by elisealix22** [reply]
Thank you. That helps tremendously! I filed an issue to track this internally and we will look into this on our Android team. Hopefully we'll have a fix for you soon.

**Reply by alcere** [reply]
Hi @elibroftw! Thanks for reporting! Please, let me clarify the issue you are experiencing:
- You visit `foss.heptapod.net`
- Then, you use the "Sign in with GitHub" option.
- You enter your GitHub credentials using your browser.
- Finally the app pops-up with that message and you are not logged in `foss.heptapod.net`

If that's the issue, can you recall since when is this happening? Can you share your OS version, the GitHub App and Firefox app versions you use? Thanks in advanced!

**Reply by elibroftw** [reply]
It's been happening for at least 2 weeks now.

Android 14. GitHub app 1.134.2. Firefox mobile 118.2.0. Here is the open by default links setting in case you do not have it set.

For GitHub:

![Screenshot_20231017-080314.png](https://github.com/community/community/assets/21298211/b56ea932-9379-4dce-86b6-0bdb4b6b33dc)

Inside the Firefox app:

![Screenshot_20231017-080404.png](https://github.com/community/community/assets/21298211/a1ea00f8-c0bf-4feb-b367-f0c05269de8b)

**Reply by elibroftw** [reply]
The issue also occurs on GitHub.com sign in

**Reply by alcere** [reply]
Got it! thanks for the details!

**Reply by stevepopovich** [reply]
Hi @elibroftw are you still having issues? We made some sign in flow changes and browsers have made some changes around deeplinks so we think this could be fixed now.

**Reply by elibroftw** [reply]
Seems to be fixed

**Reply by elibroftw** [reply]
It's not fine. Just encountered the issue today.

**Reply by stevepopovich** [reply]
Hmmm ok. Is this with `foss.heptapod.net` still?

**Reply by elibroftw** [reply]
It was with.....signing into github.com

**Reply by stevepopovich** [reply]
Oh, you can no longer sign in to github.com at all? Can you use the mobile app?

**Reply by elibroftw** [reply]
Oh I signed it alright. But I had to disable "open links in app always" in firefox. Before that I would press sign in and the app would open to something went wrong.

**Reply by stevepopovich-gh** [reply]
Ok, sorry having trouble understanding here. What exactly is the problem you are having now? Can you take a video or write repro steps?

**Reply by elibroftw** [reply]
https://photos.app.goo.gl/4CsiXZ7F9PmLUfRK8

**Reply by stevepopovich** [reply]
Just updating here: We are still looking at this internally. It is not a widespread issue and it is difficult for us to reproduce so we are trying to make sure we make the right choice here. Thanks for your patience

**Reply by markmelnic** [reply]
Seems like 2 years later this has yet to be fixed.

**Comment by ADzeranov** [comment]
ааа

**Comment by badergithub** [comment]
Yes I have the GitHub account and app

**Comment by NVolcz** [comment]
I am experiencing the same issue and have a similar setup.
Samsung S23, Firefox for Android and the GitHub app. It is happening when trying to Netlify using the OAuth signin.

**Reply by stevepopovich** [reply]
Does anything in this guide help? https://github.com/orgs/community/discussions/68753

**Comment by AngryPlayer04** [comment]
I'm also having this error, for me, after filling in the login, github asks to select a browser to open (I have 2), I click on firefox and it takes me back to the login page,getting in a loop asking again to select a browser. OneUI 5.1, Android 13, S20 FE.

**Reply by stevepopovich** [reply]
Does anything in this guide help? https://github.com/orgs/community/discussions/68753

**Reply by AngryPlayer04** [reply]
The problem was just logging in to the browser, in the app I am already logged.
What solved for me was disabling the "Open links in app".

**Reply by stevepopovich** [reply]
Gotcha, thanks for that input. We are trying to fix this, but it is very inconsistent.

**Reply by endolith** [reply]
Same problem for me. GitHub login opens a link to the GitHub login, etc.

**Reply by iskae** [reply]
Hey folks 👋 
We released a fix for this last Friday and are currently working on improving login flow. Hopefully, we will be able to provide a more stable user flow soon. 

For the login loop issues, can you please try uninstall & reinstall @endolith @AngryPlayer04 and let us know if it improved things?

Thanks in advance!

**Reply by markmelnic** [reply]
This is definitely a Firefox issue although I can't put my finger on it. Switched to developer edition for mobile and everything works perfectly.

**Reply by AngryPlayer04** [reply]
> Hey folks 👋 We released a fix for this last Friday and are currently working on improving login flow. Hopefully, we will be able to provide a more stable user flow soon.
> 
> For the login loop issues, can you please try uninstall & reinstall @endolith @AngryPlayer04 and let us know if it improved things?
> 
> Thanks in advance!

I can confirm its working as expected now, thanks!

**Reply by endolith** [reply]
I uninstalled and reinstalled the GitHub app and was then able to login to both the app and the website in Firefox

**Reply by iskae** [reply]
I am so glad that it works for you folks ❤️  Please don't hesitate to reach out if something does not work in the future 🙇
